packages/@okta/vuepress-site/guides/sign-into-native-ui/handle-authentication-responses/iOS/handle-responses.md

```diff
@@ -0,0 +1,107 @@
+
+```swift
+func handleStatus(status: OktaAuthStatus) {
+
+    switch status.statusType {
+        case .success:
+            let successState: OktaAuthStatusSuccess = status as! OktaAuthStatusSuccess
+            handleSuccessStatus(successStatus: successStatus)
+
+        case .passwordWarning:
+            let warningPasswordStatus: OktaAuthStatusPasswordWarning = status as! OktaAuthStatusPasswordWarning
+            handlePasswordWarning(passwordWarningStatus: warningPasswordStatus)
+
+        case .passwordExpired:
+            let expiredPasswordStatus: OktaAuthStatusPasswordExpired = status as! OktaAuthStatusPasswordExpired
+            handleChangePassword(passwordExpiredStatus: expiredPasswordStatus)
+
+        case .MFAEnroll,
+             .MFAEnrollActivate,
+             .MFARequired,
+             .MFAChallenge.
+             .recovery,
+             .recoveryChallenge,
+             .passwordReset,
+             .lockedOut:
+             let alert = UIAlertController(title: "Error", message: "Not implemented", preferredStyle: .alert)
+             alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+             present(alert, animated: true, completion: nil)
+    }
+}
+```
+
+### Success status handling
+
+Downcast the received status to OktaAuthStatusSuccess class and get access to `sessionToken` property. You now can exchange session token to access token via `OktaOidc` library.
+
+```swift
+let successState: OktaAuthStatusSuccess = status as! OktaAuthStatusSuccess
+let alert = UIAlertController(title: "Success", message: "Sesssion token: \(successState.sessionToken)", preferredStyle: .alert)
+alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+present(alert, animated: true, completion: nil)
+```
+
+### Error handling
+
+Errors are returned via `onError` closure. Handle `error` parameter of type `OktaError` to get the idea of error condition. `OktaError` definition if following:
+
+```swift
+public enum OktaError: Error {
+    case errorBuildingURLRequest
+    case connectionError(Error)
+    case emptyServerResponse
+    case invalidResponse
+    case responseSerializationError(Error, Data)
+    case serverRespondedWithError(OktaAPIErrorResponse)
+    case unexpectedResponse
+    case wrongStatus(String)
+    case alreadyInProgress
+    case unknownStatus(OktaAPISuccessResponse)
+    case internalError(String)
+    case invalidParameters(String)
+}
+```
+Note that some errors has underlying raw values of different types: `String`, `Error`, `OktaAPIErrorResponse` and etc. Raw values give additional context about the error and simplify error handling and troubleshooting. As an example `serverRespondedWithError` indicates API level errors and also carries server-side data encapsulated in `OktaAPIErrorResponse` class. Example below shows how to extract server-side error response:
+
+```swift
+switch error {
```

Comment:
Is this the contents of `handleError` from the previous step? Can we show the whole method?